segments/stack_project.p9k

```diff
@@ -0,0 +1,34 @@
+# vim:ft=zsh ts=2 sw=2 sts=2 et fenc=utf-8
+################################################################
+# @title powerlevel9k Segment - Haskell Stack Project
+# @source [powerlevel9k](https://github.com/bhilburn/powerlevel9k)
+##
+
+################################################################
+# Register segment
+# Parameters:
+#   segment_name  context  foreground  background  Generic  Flat/Awesome-Patched  Awesome-FontConfig  Awesome-Mapped-FontConfig  NerdFont
+#                                                                                    
+p9k::register_segment "STACK_PROJECT" "" "purple3" "white" ''  $'\u03BB='  $'\u03BB='  $'\u03BB='  $'\uE777'
+
+################################################################
+# @description
+#   This segment displays if the current directory belongs to a Haskell Stack project.
+##
+# @args
+#   $1 string Alignment - left | right
+#   $2 integer Segment index
+#   $3 boolean Whether the segment should be joined
+##
+prompt_stack_project() {
+
+    local haskellstack_version=$(stack --version 2>/dev/null | grep -oE '[0-9.]+' | head -n1)
```

Comment:
I realize there is no way to not call out to `stack`, here, but avoiding the external calls to `grep` and `head`, and instead doing that string parsing & slicing purely in ZSH, will be much faster.

We have plenty of people who can help you put together the appropriate ZSH regex, if you'd like some help!

Reply:
Here is an example:
```zsh
[[ $(echo "Version 1.7.1 x86_64") =~ "([0-9.]+)" ]] && echo "${match[1]}"
```

Reply:
Wow, I didn't know you could do that with ZSH! Fine, I'm changing that lines a bit then.